hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/cdc/InternalRowToJsonStringConverter.scala

```diff
@@ -36,15 +37,63 @@ class InternalRowToJsonStringConverter(schema: StructType) {
   }
 
   def convert(record: InternalRow): UTF8String = {
-    val map = scala.collection.mutable.Map.empty[String, Any]
+    // Use LinkedHashMap to preserve field order
+    val map = scala.collection.mutable.LinkedHashMap.empty[String, Any]
     schema.zipWithIndex.foreach {
       case (field, idx) =>
-        if (field.dataType.isInstanceOf[StringType]) {
-          map(field.name) = Option(record.getUTF8String(idx)).map(_.toString).orNull
-        } else {
-          map(field.name) = record.get(idx, field.dataType)
-        }
+        map(field.name) = convertField(record.get(idx, field.dataType), field.dataType)
     }
     UTF8String.fromString(mapper.writeValueAsString(map))
   }
+
+  private def convertField(value: Any, dataType: DataType): Any = {
+    if (value == null) {
+      null
+    } else {
+      dataType match {
+        case StringType => value.toString
```

Comment:
I see the original logic contains this step: `record.getUTF8String(idx)`, is it redundant?

Reply:
I think there is only an unit test `TestInternalRowToJsonStringConverter` covering the converting logic from internalRow to cdc json string.

Reply:
we better have a UT or e2e of Spark cdc write/read to valite it works.

Reply:
ok, will add one